### PR TITLE
Display price depend on price location

### PR DESCRIPTION
**Pull request by mynewvk**
### All Submissions:

* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
Improve wc_get_price_to_display function. Accept location as an argument to have the ability display price which uses the setting for the cart. 

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

Ability to set price_location for the wc_get_price_to_display function to respect the 'woocommerce_tax_display_cart' option.

### CODE REVIEW

File: includes/wc-product-functions.php [hunk 1]
```diff
@@ -1048,13 +1048,17 @@ function wc_get_price_to_display( $product, $args = array() ) {
 		array(
 			'qty'   => 1,
 			'price' => $product->get_price(),
+			'price_location' => 'shop'
 		)
 	);
 
 	$price = $args['price'];
 	$qty   = $args['qty'];
+	$price_location = $args['location'];
```
Comment (kloon) [inline includes/wc-product-functions.php]: In the default args above you are using `price_location`, and here you are looking for `location`.

File: includes/wc-product-functions.php [hunk 2]
```diff
@@ -1048,13 +1048,17 @@ function wc_get_price_to_display( $product, $args = array() ) {
 		array(
 			'qty'   => 1,
 			'price' => $product->get_price(),
+			'price_location' => 'shop'
```
Comment (kloon) [inline includes/wc-product-functions.php]: This default name does not match the $args call below. Also for better clarity perhaps change the args name for this to display_location

**Review by kloon — DISMISSED** [review]
A couple of small issues to fix, also could you please add some unit tests to cover all the combinations of this new argument?

**Comment by mynewvk** [comment]
Sorry about the args key - I changed it at the last moment...

File: includes/wc-product-functions.php (hunk 2 above)
Comment (mynewvk) [reply includes/wc-product-functions.php]: done

File: includes/wc-product-functions.php (hunk 1 above)
Comment (mynewvk) [reply includes/wc-product-functions.php]: done

File: includes/wc-product-functions.php [hunk 3]
```diff
@@ -1048,13 +1048,17 @@ function wc_get_price_to_display( $product, $args = array() ) {
 		array(
 			'qty'   => 1,
 			'price' => $product->get_price(),
+			'display_location' => 'shop'
```
Comment (claudiosanches) [inline includes/wc-product-functions.php]: Missing align.

It suppose to be:

```php
	$args = wp_parse_args(
		$args,
		array(
			'qty'              => 1,
			'price'            => $product->get_price(),
			'display_location' => 'shop'
		)
	);
```

File: includes/wc-product-functions.php [hunk 4]
```diff
@@ -1048,13 +1048,17 @@ function wc_get_price_to_display( $product, $args = array() ) {
...
 		)
 	);
 
 	$price = $args['price'];
 	$qty   = $args['qty'];
+	$price_location = $args['display_location'];
```
Comment (claudiosanches) [inline includes/wc-product-functions.php]: Need to align here too:

```php
	$price          = $args['price'];
	$qty            = $args['qty'];
	$price_location = $args['display_location'];
```

**Review by claudiosanches — CHANGES_REQUESTED** [review]
In general looks good, just some minor issues with coding standards and documentation, but great work!

But note that since we are now also working with `woocommerce_tax_display_cart` we should update the functions docblock that says:

```
Returns the price including or excluding tax, based on the 'woocommerce_tax_display_shop' setting.
```

Probably better change to:

```
Returns the price including or excluding tax.

By default it's based on the 'woocommerce_tax_display_shop' setting.
Use $arg['display_location'] as 'cart' to base on the 'woocommerce_tax_display_cart' setting.
```

Please let me know if you are able update it fixing the issues I reported.
Thanks.

**Comment by mynewvk** [comment]
Hey @claudiosanches, I've pushed a new commit, please let me know if it requires some other changes.

File: tests/unit-tests/tax/tax.php [hunk 5]
```diff
@@ -621,8 +631,144 @@ public function test__update_tax_rate_cities() {
...
+			$wpdb->prepare(
+				"SELECT location_code FROM {$wpdb->prefix}woocommerce_tax_rate_locations WHERE tax_rate_id = %d ORDER BY location_code ASC",
+				$tax_rate_id
+			)
+		);
 
 		$this->assertEquals( array( 'SOMEWHERE', 'SOMEWHERE_ELSE' ), $results );
 	}
+
+	public function test__price_to_display() {
```
Comment (claudiosanches) [inline tests/unit-tests/tax/tax.php:644]: Please move it to `tests/unit-tests/product/functions.php`, and rename to `test_wc_get_price_to_display()`

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Just a small tweak in the unit test and it will be ready to go.

**Comment by peterfabian** [comment]
Hi @mynewvk, would you be able to update the PR. We've refactored the unit tests a bit, so the files moved around, which should get updated as well. 
We should merge it before it gets too stale :) Thanks!

**Comment by lkraav** [comment]
I think "location" terminology is misused here. It should be "context". Location usually is short for "geolocation", and that's what I was wondering about - if this was some geolocation multi-currency thing or what.

**Review by jorgeatorres — APPROVED** [review]
LGTM.

**Pull request merged**